src/common/font.c

```diff
@@ -79,6 +79,18 @@ font_width(struct font *font, const char *string)
 	return rectangle.width;
 }
 
+void
+font_get_buffer_size(int max_width, const char *text, struct font *font,
+	int *width, int *height)
+{
+	PangoRectangle text_extents = font_extents(font, text);
+	if (max_width > 0 && text_extents.width > max_width) {
+		text_extents.width = max_width;
+	}
+	*width = text_extents.width;
```

Comment:
This looks like it could be removed completely and the assignment changed to `*width = MAX(text_extents.width, max_width)` instead.